prophy/generators.py

```diff
@@ -0,0 +1,361 @@
+from .scalar import u32
+from .data_types import base_array, build_container_length_field, distance_to_next_multiply, struct_packed
+from .desc_item import descriptor_item_type, codec_kind
+from .exception import ProphyError
+from .six import long, string_types
+
+
+class _generator_base(type):
+    """
+        Base metaclass type intended to validate, supplement and create all
+        prophy_data_object classes.
+        All of this magic happen at the very moment of the created class' import.
```

Comment:
Well, `all of this magic happens at the very moment when class definition gets interpreted, which usually happens at codec module import`. In unit tests we delay moment of class creation until fixture function is called - which happens way after import.